File: Updog.Client/src/user/common/user-login.ts
import { User } from './user';

/**
 * Login information sent back by the backend.
 */
export class UserLogin {
    /**
     * Create a new user login.
     * @param user The user of the login.
     * @param authToken Their auth token (JWT).
     */
    constructor(public user: User, public authToken: string) {}
}

File: Updog.Domain/User/ValueObjects/UserLogin.cs
using Updog.Domain;

namespace Updog.Domain {
    /// <summary>
    /// User login record and the issued auth token.
    /// </summary>
    public sealed class UserLogin : IValueObject {
        #region Properties
        /// <summary>
        /// The user it belongs to.
        /// </summary>
        public int UserId { get; }

        /// <summary>
        /// The auth token they were issued.
        /// </summary>
        /// <value></value>
        public string AuthToken { get; }
        #endregion

        #region Constructor(s)
        public UserLogin(int userId, string authToken) {
            UserId = userId;
            AuthToken = authToken;
        }
        #endregion
    }
}